metamorph/plugins/morph_pdc.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python
+import argparse
+import logging
+import logging.config
+
+import requests
+
+from metamorph.lib.logging_conf import setup_logging, storing_pretty_json
+
+
+class PDCApiException(Exception):
+    pass
+
+
+class PDCApi(object):
+    pdc_name_mapping = {
+        "bugzilla-components": {"name": '{}'},
+        "global-components": {"name": '{}'},
+        "release-component-contacts": {"component": '^{}$'},
+        "release-component-relationships": {"from_component_name": '{}'},
+        "release-components": {"name": '{}'},
+        "rpms": {"name": '^{}$', "version": '{}', "release": '{}'},
+        "global-component-contacts": {"component": '^{}$'}
+    }
+
+    pdc_param_mapping = {
+        "build-image-rtt-tests": "build_nvr",
+    }
+
+    param_mapping = {
+        "name": "",
+        "version": "",
+        "release": ""
+    }
+
+    MAX_QUERIED_DATA_SIZE = 20
+
+    def __init__(self, pdc_api_url, ca_cert, component_nvr):
+        self.pdc_api_url = pdc_api_url
+        self.ca_cert = ca_cert
+        self.component_nvr = component_nvr
+        self.pdc_proxy = None
+
+    def get_pdc_metadata_by_component_name(self, limit=10):
+        component_name, version, release = self.get_component_nvr(self.component_nvr)
+        self.setup_pdc_metadata_params(component_name, version, release)
+        logging.debug("PDC options by component name are {0} ".format(self.pdc_name_mapping))
+        logging.debug("Connecting to PDC api.")
+        pdc_metadata = {}
+        for pdc_metadata_type in self.pdc_name_mapping:
+            url = "{0}/{1}/?".format(self.pdc_api_url, pdc_metadata_type)
+            metadata = []
+            while url and len(metadata) / self.MAX_QUERIED_DATA_SIZE < limit:
+                queried_data = self.query_pdc(url, self.pdc_name_mapping[pdc_metadata_type])
+                url = queried_data['next']
+                metadata += queried_data['results']
+            pdc_metadata[pdc_metadata_type] = metadata
+        return pdc_metadata
+
+    def query_pdc(self, url, url_options=dict,):
+        """
+        This method queries pdc_api with given url and url_option variable
+
+        :param url -- pdc api url
+        :param url_options -- dictionary of wanted options
+
+        :returns -- Queried data
+        """
+        try:
+            response = requests.get(url, params=url_options, verify=self.ca_cert)
+            response.raise_for_status()
+            return response.json()
+        except requests.HTTPError as detail:
```

Comment:
Just a quick note, you can skip escaping the double quotes in python by just starting and terminating with a single quote instead. The following line could be rewritten as:
```
logging.error('Unsuccessful connection to pdc api url '
              'with url="{0}" and options="{1}"'.format(url, url_options))